### PR TITLE
Document pipeline refit on the full training set

### PR DESCRIPTION
**Pull request by matthewwritter**
## What does this PR do?
For issue #589:  Add documentation to clarify that the model generated by fit() is trained on all data, regardless of the cross-validation strategy used.

## Where should the reviewer start?
tpot/base.py  

The verbiage is identical to docs_sources/api.md

Slightly more editing was done in tpot/docs_sources/using.md

## How should this PR be tested?
Testing should not be necessary, as only documentation was edited. That said, because the phrase "testing should not be necessary" is highly correlated with impending disaster, I did run all the nosetests.

A user test could including running it by saddy001, who reported the motivating issue, and asking whether he thinks that it would have enabled him to debug the issue faster.

## Any background context you want to provide?
I decided against editing the Returns section of the doctstring, prioritizing brevity, but would certainly be open to arguments for doing so.

## What are the relevant issues?
#589 #583 

## Questions:
* Do the docs need to be updated? No further than they have been
* Does this PR add new (Python) dependencies? No

### CODE REVIEW

File: docs_sources/api.md [hunk 1]
```diff
@@ -270,7 +270,7 @@ fit(features, classes, sample_weight=None, groups=None)
 <div style="padding-left:5%" width="100%">
 Run the TPOT optimization process on the given training data.
 <br /><br />
-Uses genetic programming to optimize a machine learning pipeline that maximizes the score on the provided features and classes. Performs internal stratified k-fold cross-validaton to avoid overfitting on the provided data.
+Uses genetic programming to optimize a machine learning pipeline that maximizes the score on the provided features and target. This pipeline optimization uses internal k-fold cross-validaton to avoid overfitting on the provided data. The best pipeline is then trained on the entire set of passed samples.
```
Comment (rhiever) [inline docs_sources/api.md]: This pipeline optimization uses --> This pipeline optimization procedure uses

The best pipeline is then trained --> At the end of the pipeline optimization procedure, the best pipeline is then trained

passed samples --> provided samples

File: docs_sources/api.md [hunk 2]
```diff
@@ -720,7 +720,7 @@ fit(features, target, sample_weight=None, groups=None)
 <div style="padding-left:5%" width="100%">
 Run the TPOT optimization process on the given training data.
 <br /><br />
-Uses genetic programming to optimize a machine learning pipeline that maximizes the score on the provided features and target. Performs internal k-fold cross-validaton to avoid overfitting on the provided data.
+Uses genetic programming to optimize a machine learning pipeline that maximizes the score on the provided features and target. This pipeline optimization uses internal k-fold cross-validaton to avoid overfitting on the provided data. The best pipeline is then trained on the entire set of passed samples.
```
Comment (rhiever) [inline docs_sources/api.md]: This pipeline optimization uses --> This pipeline optimization procedure uses

The best pipeline is then trained --> At the end of the pipeline optimization procedure, the best pipeline is then trained

passed samples --> provided samples

File: docs_sources/using.md [hunk 3]
```diff
@@ -74,8 +74,8 @@ Now TPOT is ready to optimize a pipeline for you. You can tell TPOT to optimize
 pipeline_optimizer.fit(X_train, y_train)
 ```
 
-The `fit` function takes in a training data set and uses k-fold cross-validation when evaluating pipelines. It then
-initializes the genetic programming algoritm to find the best pipeline based on average k-fold score.
+The `fit` function initializes the genetic programming algorithm to find the highest-scoring pipeline based on average k-fold cross-validation
+Then, the pipeline is trained on the entire set of passed samples, and the TPOT instance can be used as a fitted model.
```
Comment (rhiever) [inline docs_sources/using.md]: passed samples --> provided samples

File: tpot/base.py [hunk 4]
```diff
@@ -457,7 +457,8 @@ def fit(self, features, target, sample_weight=None, groups=None):
 
         Uses genetic programming to optimize a machine learning pipeline that
         maximizes score on the provided features and target. Performs internal
-        k-fold cross-validaton to avoid overfitting on the training data.
+        k-fold cross-validaton to avoid overfitting on the provided data. The
+        best pipeline is then trained on the entire set of passed samples.
```
Comment (rhiever) [inline tpot/base.py]: passed samples --> provided samples

**Comment by rhiever** [comment]
I made some minor suggestions for wording. Once those are sorted out, I think we're good to merge.

**Review by weixuanfu — COMMENTED** [review]
Looks good! Thanks @matthewwritter

**Comment by matthewwritter** [comment]
OK, I've made all the changes and everything's passed. I think that's all you need from my end, but let me know if there's another step. Apologies for the latency - I mostly on get to work on this first thing in the morning or last thing at night, while the rest of the house is asleep.

**Comment by rhiever** [comment]
Merged. Thank you for the PR, @matthewwritter!

**Pull request merged**